### PR TITLE
add howto_customize_docfx_flavored_markdown.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

File: Documentation/tutorial/howto_customize_docfx_flavored_markdown.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+</code></pre>
+```
+
+### Create customize rendering plugin project
+To complete this goal, we need create a customize rendering plugin.
+
+1. Create a project, set project names.
+2. Add required nuget package: `Microsoft.DocAsCode.Dfm` with version >= 2.17.
+3. Create a class, for example with name `CustomBlockCodeRendererPart`
+4. Inhert @"Microsoft.DocAsCode.Dfm.DfmCustomizedRendererPartBase`3" (which implements @Microsoft.DocAsCode.Dfm.IDfmCustomizedRendererPart)
```
Comment (superyyrrzz) [inline Documentation/tutorial/howto_customize_docfx_flavored_markdown.md]: >Inhert [](start = 3, length = 6)

Inherit

File: Documentation/tutorial/howto_customize_docfx_flavored_markdown.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+           yield return new CustomBlockCodeRendererPart();
+       }
+   }
+   ```
+8. Build the project.
+
+### Enable customize rendering plugin
+
+1. Copy output assemblies to x\plugins, x is any folder.
+2. Run `docfx.exe` with template x ([details](howto_build_your_own_type_of_documentation_with_custom_plug-in.html#enable-plug-in))
```
Comment (superyyrrzz) [inline Documentation/tutorial/howto_customize_docfx_flavored_markdown.md]: >html [](start = 109, length = 4)

md

File: Documentation/tutorial/howto_customize_docfx_flavored_markdown.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+1. Create a new token
+2. Create a new rule
+3. Create a new renderer part
+4. Create a new renderer part provider
+5. Create a new DFM engine customizer
+
+The steps do not decrease, but you can combine any number of customize markdown extension plugin as you like.
+
+### How to create a new markdown extension by plugin
+1. Define markdown syntax (same with [markdown lite](intro_markdown_lite.html#define-markdown-syntax)).
```
Comment (superyyrrzz) [inline Documentation/tutorial/howto_customize_docfx_flavored_markdown.md]: >html [](start = 73, length = 4)

md

File: Documentation/tutorial/howto_customize_docfx_flavored_markdown.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+### Enable customize rendering plugin
+
+1. Copy output assemblies to x\plugins, x is any folder.
+2. Run `docfx.exe` with template x ([details](howto_build_your_own_type_of_documentation_with_custom_plug-in.html#enable-plug-in))
+
+Customize Markdown Extension
+----------------------------
+
+In markdown lite, we can [customize markdown extension](intro_markdown_lite.html#how-to-customize-markdown-syntax) by following steps:
```
Comment (superyyrrzz) [inline Documentation/tutorial/howto_customize_docfx_flavored_markdown.md]: >html [](start = 76, length = 4)

md

File: Documentation/tutorial/howto_customize_docfx_flavored_markdown.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
+How-to: Customize DocFX Flavored Markdown
+=========================================
+
+Customize Renderer
+------------------
+
+DocFX Flavored Markdown introduced @Microsoft.DocAsCode.Dfm.IDfmCustomizedRendererPartProvider from v2.17.
```
Comment (hellosnow) [inline Documentation/tutorial/howto_customize_docfx_flavored_markdown.md]: >introduced [](start = 24, length = 10)

is introduced?

File: Documentation/tutorial/howto_customize_docfx_flavored_markdown.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+### Set goal
+Now we want any csharp code (`cs`, `c#`, `csharp`) will generate following html:
+```html
+<pre><code class="lang-csharp">Console.WriteLine();
+</code></pre>
+```
+
+### Create customize rendering plugin project
+To complete this goal, we need create a customize rendering plugin.
```
Comment (hellosnow) [inline Documentation/tutorial/howto_customize_docfx_flavored_markdown.md]: >need create [](start = 26, length = 11)

need to create

Comment (hellosnow) [inline Documentation/tutorial/howto_customize_docfx_flavored_markdown.md]: >customize [](start = 40, length = 9)

customized

File: Documentation/tutorial/howto_customize_docfx_flavored_markdown.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+1. Create a project, set project names.
+2. Add required nuget package: `Microsoft.DocAsCode.Dfm` with version >= 2.17.
+3. Create a class, for example with name `CustomBlockCodeRendererPart`
```
Comment (hellosnow) [inline Documentation/tutorial/howto_customize_docfx_flavored_markdown.md]: >CustomBlockCodeRendererPart [](start = 42, length = 27)

minor: append dot '.' for 3 and 4?

File: Documentation/tutorial/howto_customize_docfx_flavored_markdown.md [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+5. Implements renderer part class like following:
+
+   ```cs
+   public class CustomBlockCodeRendererPart : DfmCustomizedRendererPartBase<IMarkdownRenderer, MarkdownCodeBlockToken, MarkdownBlockContext>
+   {
+       public override string Name => "MyFirstCustomRendererPart";
+   
+       public override bool Match(IMarkdownRenderer renderer, MarkdownCodeBlockToken token, MarkdownBlockContext context)
+       {
+           return token.Lang == "cs" || token.Lang == "c#";
```
Comment (hellosnow) [inline Documentation/tutorial/howto_customize_docfx_flavored_markdown.md]: >|| token.Lang == "c#" [](start = 37, length = 21)

minor: also || token.Lang == "csharp"?

File: Documentation/tutorial/howto_customize_docfx_flavored_markdown.md [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+   }
+   ```
+8. Build the project.
+
+### Enable customize rendering plugin
+
+1. Copy output assemblies to x\plugins, x is any folder.
+2. Run `docfx.exe` with template x ([details](howto_build_your_own_type_of_documentation_with_custom_plug-in.html#enable-plug-in))
+
+Customize Markdown Extension
```
Comment (superyyrrzz) [inline Documentation/tutorial/howto_customize_docfx_flavored_markdown.md]: add a short summary of this extension at the start?

File: Documentation/tutorial/howto_customize_docfx_flavored_markdown.md (hunk 5 above)
Comment (hellosnow) [inline Documentation/tutorial/howto_customize_docfx_flavored_markdown.md]: read again, original "introduced" is correct

---
In reply to: [118661879](https://github.com/dotnet/docfx/pull/1705#discussion_r118661879) [](ancestors = 118661879)

**Pull request merged**